backend/typescript/middlewares/validators/checkInValidators.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response, NextFunction } from "express";
+import { DayPart, Frequency, Status } from "../../types";
+import {
+  getApiValidationError,
+  validatePrimitive,
+  validateDate,
+  validateCategories,
+  validateRecurringDonationEndDate,
+} from "./util";
+
+export const createCheckInDtoValidator = async (
```

Comment:
can we name this be something more general if we decide to reuse for delete endpoint? or would it be better to create separate validators or these two - just a thought!

Reply:
For creating a check-in we have to verify that `startDate` and `endDate` exist in the first place, whereas they aren't required for updating check-ins. We could generalize the validator to take in a parameter (to see if it we're verifying a create dto or update dto) to account for this slight difference. Not sure exactly which is better practice @Shehryar21 @hanlinc27

Reply:
I think you can have a general validator and have another one for `create` specifically that calls it (passing in the req, res, next) while then checking `startDate` and `endDate` !